app/src/main/java/org/splitbrain/giraffe/EventLoader.java

```diff
@@ -101,25 +180,51 @@ protected String doInBackground(URL... urls) {
             //AssetManager assetManager = context.getAssets();
             //InputStream inputStream = assetManager.open("rp2011.ics");
 
-            SimpleIcalParser ical = new SimpleIcalParser(inputStream);
-            SimpleIcalEvent event;
-            while ((event = ical.nextEvent()) != null) {
-                // build record
-                EventRecord record = getEventRecord(event);
-                if (record == null) continue;
-
-                // add to database:
-                db.addEventRecord(record);
-                count++;
-
-                // progress feedback
-                publishProgress("Loaded " + count + " events...");
-
-                // abort if cancelled
-                if (isCancelled()) {
-                    db.rollback();
-                    db.close();
-                    return "Cancelled";
+            if (type.equals("ical")) {
+                SimpleIcalParser ical = new SimpleIcalParser(inputStream);
+                SimpleIcalEvent event;
+                while ((event = ical.nextEvent()) != null) {
+                    // build record
+                    EventRecord record = getEventRecordIcal(event);
+                    if (record == null) continue;
+
+                    // add to database:
+                    db.addEventRecord(record);
+                    count++;
+
+                    // progress feedback
+                    publishProgress("Loaded " + count + " events...");
+
+                    // abort if cancelled
+                    if (isCancelled()) {
+                        db.rollback();
+                        db.close();
+                        return "Cancelled";
+                    }
+                }
+            } else if (type.equals("json")) {
+                SimpleJsonParser json = new SimpleJsonParser(output);
+                ArrayList<SimpleJsonEvent> event = json.nextEvent();
+                int i;
+
+                for (i = 0; i < event.size(); i++) {
+                    // build record
+                    EventRecord record = getEventRecordJson(event.get(i));
+                    if (record == null) continue;
+
+                    // add to database:
+                    db.addEventRecord(record);
+                    count++;
+
+                    // progress feedback
+                    publishProgress("Loaded " + count + " events...");
+
+                    // abort if cancelled
+                    if (isCancelled()) {
+                        db.rollback();
+                        db.close();
+                        return "Cancelled";
+                    }
```

Comment:
Thanks for reviewing @splitbrain . Should I read the events with another unified method? I am doing this because the difference in parser and event model for iCal and JSON are different.